presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -67,6 +67,21 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
         ImmutableList.Builder<ByteCodeNode> defaultBucket = ImmutableList.builder();
         ImmutableSet.Builder<Object> constantValuesBuilder = ImmutableSet.builder();
 
+        boolean areSmallPrimitives = false;
+        if (values.size() < 1000 && javaType == long.class) {
```

Comment:
I would use <= as it is easier to communicate the optimization applying for lists "up to 1000 items long"